File: Exercism Codes/Java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String[] arr = logLine.split(":", 2);
        return arr[1].trim();
    }

    public static String logLevel(String logLine) {
        String[] arr = logLine.split(":", 2);
        return arr[0].replace("[", "").replace("]", "").toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}

File: Exercism Codes/C++/log_levels.cpp
#include <string>

namespace log_line {
	std::string message(std::string log) {
		return log.substr(log.find(":") + 2);
	}
	std::string log_level(std::string log) {
		return log.substr(1, log.find("]") - 1);
	}
	std::string reformat(std::string log) {
		return message(log) + " (" + log_level(log) + ")";
	}
}
